### PR TITLE
Add data loader customization guide as developer docs

### PR DESCRIPTION
**Pull request by kakirastern**
## Description
To add developer docs for a guide to add a customized data loader in the `glue-solar` plugin to load custom data to `glue`.

### CODE REVIEW

File: docs/dev-docs-for-loader-customization.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
+.. _loader_customization_guide:
+
+===============================
+Data Loader Customization Guide
+===============================
+
+The `glue` data visualization tool is highly customizable for different data types, such as for FITS files
+that are often used in solar physics research. There are two ways to customize `glue` using the `glue-solar` plugin:
```
Comment (nabobalis) [inline docs/dev-docs-for-loader-customization.rst]:
```suggestion
that are often used in solar physics research.
There are two ways to customize `glue` using the `glue-solar` plugin:
```

File: docs/dev-docs-for-loader-customization.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
+.. _loader_customization_guide:
+
+===============================
+Data Loader Customization Guide
+===============================
+
+The `glue` data visualization tool is highly customizable for different data types, such as for FITS files
```
Comment (nabobalis) [inline docs/dev-docs-for-loader-customization.rst]:
```suggestion
The `glue` data visualization tool is highly customizable for different data types, such as FITS files
```

File: docs/dev-docs-for-loader-customization.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+that are often used in solar physics research. There are two ways to customize `glue` using the `glue-solar` plugin:
+
+1. The first option is to develop a custom loader for the solar data taken from practically any (ground-based or
+satellite-borne) instrument.
+
+2. The second option is to develop a custom viewer to enable a certain very specific way to view your data, such as
```
Comment (nabobalis) [inline docs/dev-docs-for-loader-customization.rst]:
```suggestion
2. The second option is to develop a custom viewer to enable a specific way to view your data, such as
```

File: docs/dev-docs-for-loader-customization.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+an alternative of the existing 2D image viewer that presents your data in a manner you like; e.g. color-coding the
```
Comment (nabobalis) [inline docs/dev-docs-for-loader-customization.rst:15]: Do we have a better example?

File: docs/dev-docs-for-loader-customization.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+lines or data points.
+
+While these two options serve different needs, the former is often needed to load your pre-processed data, as `glue`
```
Comment (nabobalis) [inline docs/dev-docs-for-loader-customization.rst]: pre-processed?

File: docs/dev-docs-for-loader-customization.rst [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+already has the functionality to display the typical FITS file (with extensions ``.fits``, ``.fit``, or ``.ft``), such
+as the existing basic 2D image viewer and 1D profile viewer (useful for viewing 1D spectra). Hence in this guide, we
+walk through how to construct such a custom data loader for your solar physics use cases.
```
Comment (nabobalis) [inline docs/dev-docs-for-loader-customization.rst]:
```suggestion
walk through how to construct such a custom data loader for your solar physics use case.
```

File: docs/dev-docs-for-loader-customization.rst [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+The Case of the IRIS subpackage
+-------------------------------
+
+If we look at the ``glue_solar/instruments/iris`` directory, we can find (at least at the time of writing of this
+guide) the files within follows:
+
+1. __init__.py: This is the typical ``init`` file that is required for running unit tests with ``pytest``.
```
Comment (nabobalis) [inline docs/dev-docs-for-loader-customization.rst]: I am not sure this is useful information?

File: docs/dev-docs-for-loader-customization.rst [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+IRIS version as a template, and then use Qt's Designer application to edit it to suit once needs.
+
+5. Some module akin to the ``utils.py`` for storing all the helper functions one will need in the other modules within
+the same instrument subpackage.
+
+Basically the most important thing to keep in mind is to make sure your main instrument module (``iris.py`` in the
+case of the ``IRIS`` satellite) contains three essential ingredients; they are namely the parser (note the use of the
+``@qglue_parser`` decorator for this), the data factory (note the use of the ``@data_factory`` decorator), and the
+importer (note the use of the ``@importer`` decorator).
+
```
Comment (nabobalis) [inline docs/dev-docs-for-loader-customization.rst:71]: I guess my issue with this part of the doc is that I don't feel like I understand more on how to write a custom loader. I know a lot how to layout a new subpackage.

File: docs/dev-docs-for-loader-customization.rst (hunk 5 above)
Comment (kakirastern) [reply docs/dev-docs-for-loader-customization.rst]: Yup sorry, another way of saying the data should first be reduced, which should be a given...

File: docs/dev-docs-for-loader-customization.rst (hunk 7 above)
Comment (kakirastern) [reply docs/dev-docs-for-loader-customization.rst]: It's chiefly for those who are new to testing with the `pytest` framework

File: docs/dev-docs-for-loader-customization.rst (hunk 8 above)
Comment (kakirastern) [reply docs/dev-docs-for-loader-customization.rst:71]: That's why I am providing the link to the official info in the end... Or do you reckon I should move this to top?

File: docs/dev-docs-for-loader-customization.rst (hunk 4 above)
Comment (kakirastern) [reply docs/dev-docs-for-loader-customization.rst:15]: Not for solar physics not at the moment...

File: docs/dev-docs-for-loader-customization.rst (hunk 7 above)
Comment (nabobalis) [reply docs/dev-docs-for-loader-customization.rst]: An init file isn't only for pytest tho.

File: docs/dev-docs-for-loader-customization.rst (hunk 5 above)
Comment (nabobalis) [reply docs/dev-docs-for-loader-customization.rst]: Would that not be post-processed?

File: docs/dev-docs-for-loader-customization.rst (hunk 4 above)
Comment (nabobalis) [reply docs/dev-docs-for-loader-customization.rst:15]: That is fine.

File: docs/dev-docs-for-loader-customization.rst (hunk 8 above)
Comment (nabobalis) [reply docs/dev-docs-for-loader-customization.rst:71]: I guess I am wondering is there more to be said from the solar physics side or you experience of writing the code for glue-solar that is missing from the official guide?

File: docs/dev-docs-for-loader-customization.rst (hunk 7 above)
Comment (kakirastern) [reply docs/dev-docs-for-loader-customization.rst]: True... Let me add to it

File: docs/dev-docs-for-loader-customization.rst (hunk 8 above)
Comment (kakirastern) [reply docs/dev-docs-for-loader-customization.rst:71]: Don't think there is much except the code to load the solar data is highly dependent on the instrument used...

File: docs/dev-docs-for-loader-customization.rst (hunk 5 above)
Comment (kakirastern) [reply docs/dev-docs-for-loader-customization.rst]: Oh, you are right

File: docs/dev-docs-for-loader-customization.rst (hunk 8 above)
Comment (nabobalis) [reply docs/dev-docs-for-loader-customization.rst:71]: Good point.

**Comment by kakirastern** [comment]
Changes made, PR ready for another review

File: docs/dev-docs-for-loader-customization.rst (hunk 8 above)
Comment (kakirastern) [reply docs/dev-docs-for-loader-customization.rst:71]: Cool 😎 Code added for this

File: docs/dev-docs-for-loader-customization.rst [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+5. Some module akin to the ``utils.py`` for storing all the helper functions one will need in the other modules within
+the same instrument subpackage.
+
+Basically the most important thing to keep in mind is to make sure your main instrument module (``iris.py`` in the
+case of the ``IRIS`` satellite) contains three essential ingredients; they are namely the parser (note the use of the
+``@qglue_parser`` decorator for this), the data factory (note the use of the ``@data_factory`` decorator), and the
+importer (note the use of the ``@importer`` decorator). Please note also that the way to convert instrument-specific
+data to the ``glue.core.data.Data`` object is highly dependent on the pipeline used for such observations. But in
+general the ``astropy.io.fits` methods should be able to handle most but not all as an instrument-agnostic option.
```
Comment (nabobalis) [inline docs/dev-docs-for-loader-customization.rst]:
```suggestion
general the ``astropy.io.fits`` methods should be able to handle most but not all as an instrument-agnostic option.
```

**Comment by kakirastern** [comment]
Thanks for catching the bug for me 👍

**Review by nabobalis — APPROVED** [review]

**Comment by kakirastern** [comment]
Thanks again @nabobalis

**Pull request merged**